stata_kernel/kernel.py

```diff
@@ -24,6 +24,7 @@ def __init__(self, *args, **kwargs):
 
         self.sc_delimit_mode = False
         self.stata = StataSession()
+        self.completions = CompletionsManager(self)
```

Comment:
No, actually. In testing this doesn't save the kernel methods inside completions. Kernel is a separate object. `self.completions.stata.do('code')` shouldn't work.

Reply:
Ok. I'd never imagined adding self as an argument to an object before.
So the benefit of this is that `CompletionsManager` has access to the methods in `StataKernel`, right?

Reply:
Yes. Some functions in `CompletionsManager` have `kernel` as an argument, so it can access the kernel methods via that argument. However, it never sales `kernel` in self, to hopefully avoid the recursion.